### DISCUSSION TITLE
HTTP X-Forwarded-Proto header security

### DISCUSSION THREAD

**Discussion by ElginCahangirov** [nginx-proxy/nginx-proxy#1746] [General]
My question is about mapping declared in nginx conf file:

```
# If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
# scheme used to connect to this server
map $http_x_forwarded_proto $proxy_x_forwarded_proto {
  default $http_x_forwarded_proto;
  ''      $scheme;
}
```

Here, as comment also says, if we receive X-Forwarded-Proto, pass it through. Doesn't this create vulnerability in web server? I mean if client sends request to `http://example.com` but manually sets header `X-Forwarded-Proto: https` we are passing `https` to the proxied application?

**Comment by buchdag** [comment]
Hi. Could you elaborate on what would be the server side vulnerability in this case ?
